portfolio/src/main/webapp/request.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * Models the requests sent to the algorithm for scheduling. This class is
+ * useful when the request information needs to be sent to the scheduling
+ * servlet through a POST request. This class contains all the request
+ * information that can be easily converted into a JSON string.
+ */
+class ScheduleRequest {
+  /**
+   * All fields are required for this object.
+   * @param events: events happening the day of the scheduling,
```

Comment:
Consider clarifying the type of events and tasks. Are they Javascript arrays of * ?

Reply:
Done.